docs: distributed Coordinator (#396)
edgelesssys/marblerun: docs/docs/features/recovery.md

With MarbleRun, the Coordinator [manages the Marbles' secrets](../features/secrets-management.md) and Marbles obtain them securely on start.
Thus, Marbles can be distributed and rescheduled on arbitrary machines.
This narrows the challenge of persistent storage down to the Coordinator itself.

### Single Coordinator

The straightforward way to run MarbleRun is with a single Coordinator.
In this case, the state is encrypted with the SGX seal key and stored on disk.

> When being pinned to a single host, the Coordinator can unseal its state automatically.

Reviewer: ```suggestion
When pinned to a single host, the Coordinator can unseal its state automatically.
```

> However, when the Coordinator is moved to another physical host, a [manual step](#recovery) is required to ensure the Coordinator's state can be recovered.

Reviewer: ```suggestion
However, when the Coordinator is moved to another physical host, a [manual step](#recovery) is required to recover the Coordinator's state.
```

### Distributed Coordinator

<enterpriseBanner/>

When you use MarbleRun [with Kubernetes](../deployment/kubernetes.md), you can scale the Coordinator to multiple instances.

> The instances share a common state, which is encrypted and stored as a Kubernetes secret.

Reviewer: ```suggestion
The instances share a common state, encrypted and stored as a Kubernetes secret.
```
